pandas/_libs/tslib.pyx

```diff
@@ -429,17 +429,32 @@ def array_with_unit_to_datetime(
             fvalues = iresult.astype('f8') * m
             need_to_iterate = False
 
+        # GH20445
+        if values.dtype.kind == "f":
+            fresult = values.astype('f8', casting='same_kind', copy=False)
+            # fill by comparing to NPY_NAT constant
+            mask = fresult == NPY_NAT
```

Comment:
why NPY_NAT here?  i would expect np.nan to  be relevant

Reply:
Right, it doesn't make much sense - looked up the definition here:
https://github.com/pandas-dev/pandas/blob/42fd7e7d9a2c115af9a52f7f896d48b75a271efe/pandas/_libs/tslibs/util.pxd#L48
`NPY_NAT` is used like this throughout the function, though. Is it worth looking into replacing it with `np.nan` (here or in a separate PR)?

Reply:
you can check for np.nan here is appropriate for a missing value (we do this elsewhere)

Reply:
ok! 

do we want to switch to np.nan just up here or everywhere in the function? There are a fair number of `NPY_NAT` uses later on:
https://github.com/pandas-dev/pandas/blob/04e9e0afd476b1b8bed930e47bf60ee20fa78f38/pandas/_libs/tslib.pyx#L264-L354